docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,488 @@
+ensuring correct execution order, and synchronous calls are necessary for
+maintaining data consistency.
+
+Synchronous calls
+-------------------------------------------------------------------------------
+
+Synchronous calls ensure task completion before moving to the next operation.
+For example, :cpp:func:`hipMemcpy` for data transfers waits for completion
+before returning control to the host. Similarly, synchronous kernel launches
+are used when immediate completion is required. When a synchronous function is
+called, control is not returned to the host thread before the device has
+completed the requested task. The behavior of the host thread—whether to yield,
+block, or spin—can be specified using :cpp:func:`hipSetDeviceFlags` with
+appropriate flags. Understanding when to use synchronous calls is important for
+managing execution flow and avoiding data races.
+
+Events for synchronization
+-------------------------------------------------------------------------------
+
+By creating an event with :cpp:func:`hipEventCreate` and recording it with
+:cpp:func:`hipEventRecord`, developers can synchronize operations across
+streams, ensuring correct task execution order. :cpp:func:`hipEventSynchronize`
+lets the application wait for an event to complete before proceeding with the next
+operation.
+
+Programmatic dependent launch and synchronization
+-------------------------------------------------------------------------------
+
+While CUDA supports programmatic dependent launches allowing a secondary kernel
+to start before the primary kernel finishes, HIP achieves similar functionality
+using streams and events. By employing :cpp:func:`hipStreamWaitEvent`, it is
+possible to manage the execution order without explicit hardware support. This
+mechanism allows a secondary kernel to launch as soon as the necessary
+conditions are met, even if the primary kernel is still running.
+
+Example
+-------------------------------------------------------------------------------
+
+The examples shows the difference between sequential, asynchronous calls and
+asynchronous calls with ``hipEvents``.
+
+.. figure:: ../../data/how-to/hip_runtime_api/asynchronous/sequential_async_event.svg
+  :alt: Compare the different calls
+  :align: center
+
+The example codes
+
+.. tab-set::
+
+    .. tab-item:: Sequential
+
+      .. code-block:: cpp
+
+        #include <hip/hip_runtime.h>
+        #include <vector>
+        #include <iostream>
+
+        #define HIP_CHECK(expression)                \
+        {                                            \
+            const hipError_t status = expression;    \
+            if(status != hipSuccess){                \
+                    std::cerr << "HIP error "        \
+                        << status << ": "            \
+                        << hipGetErrorString(status) \
+                        << " at " << __FILE__ << ":" \
+                        << __LINE__ << std::endl;    \
```

Comment:
maybe add an `exit` or `std::abort` here if check fails.

Reply:
I would avoid the ``exit`` usage. We mentioned this at the error handling page:
https://rocm.docs.amd.com/projects/HIP/en/docs-develop/how-to/hip_runtime_api/error_handling.html#hip-check-macros